components/base-war/src/main/webapp/resources/uicomponents/widgets/userpicker/userPicker.js

```diff
@@ -120,6 +120,8 @@ widgets.UserPicker = Class.create(widgets.Suggest, {
 
   // @Override
   createItemDisplay: function(data, source) {
+    data.info = data.info.escapeHTML();
```

Comment:
I'm not a fan of this kind of mutation, but its about as close as I can think of escaping before element insertion (without affecting other logic). We have to escape `sInput` right before the `emphasizeMatches()` so that the html tags can be added in.

Still cannot merge this PR as I'm still trying to fix the injection via the initial creation of Work Group. I think the document name is not being escaped.

Reply:
There are two preferred locations for escaping:

1. ASAP, when the data is first obtained from the database; in this case it would be in the code that serves the suggestions, serverside
2. ALAP, when the data is placed in the output

The first one has the advantage of being somewhat safer, since it's only one place, so there's no risk of someone later forgetting to escape. But it has the big disadvantage that only one kind of escape is allowed, and in our platform data is usually used in multiple different ways: SQL, JSON, HTML, JavaScript, so it doesn't really work for us.

The second one is more semantically correct, since you escape right where the variable is inserted in content, and the escape method is obvious. The problem is that it is very easy to forget to apply the right escape, which may lead to different kinds of injections.

An escape in between these points makes it difficult to comprehend the code, since one must read all the code to figure out if some variable has already been escaped or not. Oother kinds